processor/resourcedetectionprocessor/factory.go

```diff
@@ -178,6 +178,7 @@ func (f *factory) getResourceDetectionProcessor(
 	}
 
 	return &resourceDetectionProcessor{
+		config:             oCfg,
```

Comment:
Do we need to pass the whole config or can we just pass the middleware id? Similar to what we do for override/httpClientSettings below